File: JavaScript/프로그래머스/2/42747. H－Index/H－Index.js
function solution(citations) {
    var answer = citations.length;
    citations.sort((a,b) => b-a);
    let i = citations[0];
    while (i >= 0) {
        let h_len = citations.filter(n => n >= i).length;
        if (h_len >= i /*&& r_max <= val*/) return i;
        
        i--;
    }
    return answer;
}

File: Java/프로그래머스/2/42747. H－Index/H－Index.java
// import java.util.*;

// class Solution {
//     public int solution(int[] citations) {
//         int answer = 0;
//         int len = citations.length;
//         Arrays.sort(citations);
        
//         for (int i = len - 1; i >= 0; i--) {
//             int hIndex = citations[i];
            
//             int count = (int)Arrays.stream(citations)
//                             .filter(v -> v >= hIndex)
//                             .count();
            
//             if (count >= hIndex && len - count <= hIndex) {
//                 answer = Math.max(len- (i+1)-1, count);
//                 break;
//             }
//         }
        
//         return answer;
//     }
// }
import java.util.Arrays;

class Solution {
    public int solution(int[] citations) {
        int answer = 0;

        Arrays.sort(citations); // 오름차순 정렬

        for(int i = citations.length - 1; i >= 0; i--) {
            if(citations.length - i >= citations[i]) break;
            answer++;
        }

        return answer;
    }
}